packages/target-tauri/src-tauri/src/webxdc.rs

```diff
@@ -53,10 +53,19 @@ pub(crate) fn webxdc_icon_protocol<R: tauri::Runtime>(
         // workaround for not yet available try_blocks feature
         // https://doc.rust-lang.org/beta/unstable-book/language-features/try-blocks.html
         let result: anyhow::Result<()> = async {
+            let parsed = {
+                #[cfg(not(any(target_os = "windows", target_os = "android")))]
+                {
+                    (request.uri().host(), request.uri().path().split('/').nth(1))
+                }
+                #[cfg(any(target_os = "windows", target_os = "android"))]
+                {
+                    let mut splited = request.uri().path().split('/');
+                    (splited.nth(1), splited.next())
+                }
+            };
```

Comment:
This code is the same as in `blobs.rs` no?

Reply:
might be, but it contains different data (blobs have `accountfolder/blobfilename` (which might need to change to accommodate for dedup), and webxdc-icons have `account-id/message-id`).

Also don't want to change anything in this file if possible, because I have a local branch where I'm working on webxdc - I want to avoid adding extra merge conflicts.